src/App.tsx

```diff
@@ -1,27 +1,70 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [hasError, setHasError] = useState(false);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const getCallback = async (callBackFunc: () => Promise<Good[]>) => {
+    const goodsFromApi = await callBackFunc();
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+    return goodsFromApi;
+  };
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+  const handleGoods = async (promiseFunc: () => Promise<Good[]>) => {
+    try {
```

Comment:
Filter pressing the same button twice to avoid loading the same data multiple times. Only in case there is no error.